docker/witnet-rust/wip0010-recovery.sh

```diff
@@ -0,0 +1,153 @@
+
+# Just a pretty logging helper
+function log {
+  echo "[WIP0010_RECOVERY] $1"
+}
+
+# A helper for calculating ETAs
+function eta {
+  START=$1
+  NOW=$2
+  PROGRESS=$3
+  if [ "$PROGRESS" == "00" ]; then
+    echo "will be shown as synchronization moves forward..."
+  else
+    ELAPSED=$(( NOW - START ))
+    SPEED=$((PROGRESS * 1000 / ELAPSED))
+    REMAINING_PROGRESS=$(( 10000 - PROGRESS ))
+    REMAINING_TIME=$((REMAINING_PROGRESS / (SPEED / 1000) ))
+    echo $(( REMAINING_TIME / 60 )) minutes $((REMAINING_TIME % 60)) seconds
+  fi
+}
+
+# This script can be skipped by setting environment variable SKIP_WIP0010_RECOVERY to "true"
+if [[ "$SKIP_WIP0010_RECOVERY" == "true" ]]; then
+  log "Skipping WIP-0010 recovery"
+  exit 0
+fi
+
+# Make sure the arguments make sense
+if ! command -v "$WITNET_BINARY" &> /dev/null; then
+  log "ERROR: The provided witnet binary (first argument to this script) is not a valid executable file: $WITNET_BINARY"
+  exit 1
+fi
+if [ ! -f "$WITNET_CONFIG_FILE" ]; then
+  log "ERROR: The provided witnet configuration file (second argument to this script) is not a valid configuration file: $WITNET_CONFIG_FILE"
+  exit 2
+fi
+
+# Read configuration (e.g. node server address) from config file
+log "Using configuration file at $WITNET_CONFIG_FILE"
+HOST=$(grep "server_addr" "$WITNET_CONFIG_FILE" | sed -En "s/server_addr = \"(.*)\"/\1/p" | sed -En "s/0.0.0.0/127.0.0.1/p" )
```

Comment:
This evaluates to empty string when the `witnet.toml` has a line like

```
server_addr = "127.0.0.1:21337"
```

The first `sed` works, but the second one doesn't.

```
$ echo 'server_addr = "127.0.0.1:21337"' | sed -En "s/server_addr = \"(.*)\"/\1/p"
127.0.0.1
$ echo 'server_addr = "127.0.0.1:21337"' | sed -En "s/server_addr = \"(.*)\"/\1/p" | sed -En "s/0.0.0.0/127.0.0.1/p"
```

Reply:
Hahahaha big F, it should be like this:
```console
echo 'server_addr = "4.0.0.0:21337"' | sed -En "s/server_addr = \"(.*)\"/\1/p" | sed -E "s/0\.0\.0.\0/127.0.0.1/g"
```